app/models/embedded_ansible_worker/runner.rb

```diff
@@ -60,13 +60,21 @@ def do_exit(*args)
     Thread.exit
   end
 
-  def update_embedded_ansible_manager
-    ansible = ManageIQ::Providers::EmbeddedAnsible::AutomationManager.first_or_initialize
-    server  = MiqServer.my_server(true)
-    ansible.default_endpoint.url = URI::HTTPS.build(:host => server.hostname, :path => "/ansibleapi/v1")
-    ansible.name = "Embedded Ansible"
-    ansible.zone = server.zone
-    ansible.save!
+  def update_embedded_ansible_provider
+    server   = MiqServer.my_server(true)
+    provider = ManageIQ::Providers::EmbeddedAnsible::Provider.first_or_initialize
+
+    provider.name = "Embedded Ansible"
```

Comment:
I'm not sure what we would gain by using a block. `.first_or_initialize` only yields if the object was not found. I might be missing something thoug, can you elaborate?